elide-datastore/elide-datastore-hibernate/src/test/java/com/yahoo/elide/datastores/hibernate/hql/RootCollectionFetchQueryBuilderTest.java

```diff
@@ -128,6 +129,56 @@ public void testRootFetchWithJoinFilter() {
         assertEquals(expected, actual);
     }
 
+    @Test
+    public void testDistinctRootFetchWithToManyJoinFilterAndPagination() {
```

Comment:
Can we test the exception behavior as well?

Reply:
yes, there is/are failing tests right now because of this, but I am struggling little bit with running and debugging them locally. when I resolve that, I'll try to write test for this case